### PR TITLE
✈️ Always set `sent_on` for W/S/D statuses if not already set

### PR DESCRIPTION
**Pull request by rowanseymour**

### CODE REVIEW

File: backends/rapidpro/backend_test.go [hunk 1]
```diff
@@ -480,16 +483,26 @@ func (ts *BackendTestSuite) TestMsgStatus() {
 	m, err := readMsgFromDB(ts.b, courier.NewMsgID(10001))
 	ts.NoError(err)
 	ts.Equal(m.Status_, courier.MsgWired)
-	ts.Equal(m.ExternalID_, null.String("ext0"))
 	ts.True(m.ModifiedOn_.After(now))
 	ts.True(m.SentOn_.After(now))
 
-	// update by id, no external id, shouldn't overwrite it
```
Comment (rowanseymour) [inline backends/rapidpro/backend_test.go:487]: some of these comments don't match what the test is doing..

**Review by nicpottier — APPROVED** [review]

File: backends/rapidpro/status.go [hunk 2]
```diff
@@ -265,11 +265,11 @@ UPDATE msgs_msg SET
 		END,
 	sent_on = CASE 
 		WHEN 
-			s.status = 'W' 
+			s.status IN ('W', 'S', 'D') AND sent_on IS NULL
```
Comment (norkans7) [inline backends/rapidpro/status.go:268]: what about the msgs with R? we clear the sent on on them?

Comment (rowanseymour) [reply backends/rapidpro/status.go:268]: we don't use `R` any more - resending requeues messages as QUEUED

Comment (norkans7) [reply backends/rapidpro/status.go:268]: never mind we use pending on them

**Review by norkans7 — APPROVED** [review]

**Pull request merged**